reference/commands/report/diff.rst

```diff
@@ -0,0 +1,77 @@
+.. _reference_commands_report_diff:
+
+conan report diff
+=================
+
+.. autocommand::
+    :command: conan report diff -h
+
+The ``conan report diff`` command gets the differences between two recipes, also comparing their sources.
+This functionality allows you to compare either two versions of the same recipe or two entirely different recipes.
+Each recipe (old and new) can be identified in one of two ways: by providing both the path to its ``conanfile.py`` and
+its reference, or by specifying just the reference.
+
+When only a reference is given, Conan will first search for the recipe in the local cache; if it is not found, it will
+attempt to download it from the configured remotes. If no revision is explicitly provided, Conan will
+default to using the latest available revision.
+
+
+**Examples**
+
+
+If we want to compare versions 1.0 and 2.0 of "mylibrary" that are available on our `my-remote` remote, it would be:
+
+.. code-block:: bash
+
+    $ conan report diff --old-reference="mylib/1.0" --new-reference="mylib/2.0" -r=my-remote
+
+
+This is how it would look if one of the versions is on the remote and the other is not. The version that is not on the
+remote requires the path to the recipe in order to compare it.
+
+.. code-block:: bash
+
+    $ conan report diff -or="mylib/1.0" --old-path="path/to/recipe" -nr="mylib/2.0"
+    $ conan report diff -or="mylib/1.0" -nr="mylib/2.0" --new-path="path/to/recipe"
+    $ conan report diff -or="mylib/1.0" --old-path="path/to/recipe" -nr="mylib/2.0" --new-path="path/to/recipe"
```

Comment:
this is a bit confusing, and also mixing abbreviated arguments with the ones that are not, can you elaborate a bit more on the previous paragraphs about each of the arguments and how they should be used please?